src/client/components/collection/CopyButton.js

```diff
@@ -0,0 +1,139 @@
+import React, { useContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import IconButton from '@material-ui/core/IconButton';
+import FileCopyIcon from '@material-ui/icons/FileCopy';
+import Tooltip from '@material-ui/core/Tooltip';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { makeStyles } from '@material-ui/core';
+import { toast } from 'react-toastify';
+import { useTranslation } from 'react-i18next';
+import {
+  COPY_ROUTE,
+  IS_AUTHENTICATED_ROUTE,
+  buildSpaceRoute,
+} from '../../config/routes';
+import { DEFAULT_GET, DEFAULT_POST } from '../../../api/common';
+import LoginModal from './LoginModal';
+import TreeModal from './TreeModal';
+import { CollectionContext } from '../CollectionProvider';
+
+const useStyles = makeStyles((theme) => ({
+  successLink: {
+    display: 'block',
+    fontStyle: 'italic',
+    color: 'inherit',
+    paddingTop: theme.spacing(0.5),
+  },
+}));
+
+const CopyButton = ({ id }) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const {
+    current: { id: collectionId },
+  } = useContext(CollectionContext);
+  const [showLoginModal, setShowLoginModal] = useState(false);
+  const [showTreeModal, setShowTreeModal] = useState(false);
+  const [isSignedIn, setIsSignedIn] = useState(false);
+  const [copying, setCopying] = useState(false);
+
+  const checkUserIsSignedIn = async () => {
+    const res = await (await fetch(IS_AUTHENTICATED_ROUTE, DEFAULT_GET)).json();
+    setIsSignedIn(res);
+  };
+
+  useEffect(async () => {
```

Comment:
I really like this!

Reply:
I think and in order to use an async call inside you effect. you should be doing something like this:
`useEffect(() => {`
  `  (async function checkSignIn() {`
      `await checkUserIsSignedIn();`
  `  })();`
`  }, []);`

check this: https://github.com/facebook/react/issues/14326#issuecomment-441680293